### ISSUE TITLE
manual_range_patterns: False positive with range patterns starting with `|`

### ISSUE THREAD

**Issue by tirr-c** [rust-lang/rust-clippy#11825]
### Summary

Range patterns with a leading `|`, without any other patterns combined with OR pattern, trigger `manual_range_patterns`.

### Lint Name

manual_range_patterns

### Reproducer

I tried [this code](https://github.com/tirr-c/jxl-oxide/blob/687cd9b0694bfd06526995ed3ecbe8d8b93dc2bc/crates/jxl-color/src/icc.rs#L63-L74):

```rust
    let p1 = match b1 {
        | b'a'..=b'z'
        | b'A'..=b'Z' => 0,
        | b'0'..=b'9'
        | b'.'
        | b',' => 1,
        | 0..=1 => 2 + b1 as u32,
        | 2..=15 => 4,
        | 241..=254 => 5,
        | 255 => 6,
        | _ => 7,
    };
```

I saw [this happen](https://github.com/tirr-c/jxl-oxide/actions/runs/6894797142/job/18757393400):

```
error: this OR pattern can be rewritten using a range
  --> crates/jxl-color/src/icc.rs:69:9
   |
69 |         | 0..=1 => 2 + b1 as u32,
   |         ^^^^^^^ help: try: `0..=1`
   |

error: this OR pattern can be rewritten using a range
  --> crates/jxl-color/src/icc.rs:70:9
   |
70 |         | 2..=15 => 4,
   |         ^^^^^^^^ help: try: `2..=15`
   |

error: this OR pattern can be rewritten using a range
  --> crates/jxl-color/src/icc.rs:71:9
   |
71 |         | 241..=254 => 5,
   |         ^^^^^^^^^^^ help: try: `241..=254`
   |
```

I think it shouldn't be treated as an OR pattern.

### Version

```text
rustc 1.74.0 (79e9716c9 2023-11-13)
binary: rustc
commit-hash: 79e9716c980570bfd1f666e3b16ac583f0168962
commit-date: 2023-11-13
host: x86_64-unknown-linux-gnu
release: 1.74.0
LLVM version: 17.0.4
```

**Comment by Centri3** [comment]
We could just ignore any `Or` patterns with only 1 element.

Is this a stylistic choice or does it actually change the meaning of the code?

**Comment by tirr-c** [comment]
The leading `|` doesn't mean anything, it's purely stylistic.

**Comment by alex-semenyuk** [comment]
Fmt is removed this OR at start so suppose it doesn't make sense to fix it on clippy side
